webapp/src/main/java/com/box/l10n/mojito/service/ai/translation/AITranslateJob.java

```diff
@@ -0,0 +1,257 @@
+  boolean isReuseSourceOnLanguageMatch;
+
+  @Override
+  @Timed("AITranslateJob.call")
+  public Void call(AITranslateJobInput input) throws Exception {
+
+    try {
+      Repository repository =
+          repositoryRepository
+              .findById(input.getRepositoryId())
+              .orElseThrow(
+                  () ->
+                      new AITranslateJobException(
+                          "Repository with id " + input.getRepositoryId() + " not found."));
+
+      TmTextUnitPendingMT pendingMT =
+          tmTextUnitPendingMTRepository.findByTmTextUnitId(input.getTmTextUnitId());
+
+      if (pendingMT != null) {
+        if (!isExpired(pendingMT)) {
+          tmTextUnitRepository
+              .findById(input.getTmTextUnitId())
+              .ifPresent(
+                  tmTextUnit -> {
+                    if (aiTranslationTextUnitFilterService.isTranslatable(tmTextUnit, repository)) {
+                      translateLocales(input, tmTextUnit, repository);
+                    } else {
+                      logger.debug(
+                          "Text unit with name: {} should not be translated, skipping AI translation.",
+                          tmTextUnit.getName());
+                      meterRegistry.counter(
+                          "AITranslateJob.translate.notTranslatable",
+                          Tags.of("repository", repository.getName()));
+                    }
+                  });
+          meterRegistry
+              .timer("AITranslateJob.timeToMT", Tags.of("repository", repository.getName()))
+              .record(Duration.between(JSR310Migration.dateTimeNow(), pendingMT.getCreatedDate()));
+        } else {
+          // If the pending MT is expired, log an error and delete it
+          logger.error("Pending MT for tmTextUnitId: {} is expired", input.getTmTextUnitId());
+          meterRegistry.counter(
+              "AITranslateJob.expired", Tags.of("repository", repository.getName()));
+        }
+      } else {
+        logger.warn("No pending MT found for tmTextUnitId: {}", input.getTmTextUnitId());
+        meterRegistry.counter(
+            "AITranslateJob.noPendingMTFound", Tags.of("repository", repository.getName()));
+      }
+    } catch (Exception e) {
+      logger.error("Error running job for text unit id {}", input.getTmTextUnitId(), e);
+      meterRegistry.counter(
+          "AITranslateJob.error", Tags.of("repository", input.getRepositoryId().toString()));
+    } finally {
+      TmTextUnitPendingMT pendingMT =
+          tmTextUnitPendingMTRepository.findByTmTextUnitId(input.getTmTextUnitId());
+      if (pendingMT != null) {
+        logger.debug("Deleting pending MT for tmTextUnitId: {}", input.getTmTextUnitId());
+        tmTextUnitPendingMTRepository.delete(pendingMT);
+      }
+    }
+    return null;
+  }
+
+  private void translateLocales(
+      AITranslateJobInput input, TMTextUnit tmTextUnit, Repository repository) {
```

Comment:
Is `AITranslateJobInput` needed here? We already are passing in the fully hydrated TMTextUnit and Repository objects - which are then a superset of AITranslateJobInput.